### ISSUE TITLE
"Mark as done" button is grayed out and appears pressed when clicking on "To" to assign a task

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#20852]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open a chat
2. Navigate to the "Assign Task" section of the chat
3. Enter the task title and click the "Next" button
5. Confirm the task by clicking "Confirm task."
6. Click on "To" to assign the task

## Expected Result:
 The "Mark as done" button should not be grayed out when clicking on "To" to assign a task. It should not appear pressed either

## Actual Result:
When clicking on "To" to assign a task, the "Mark as done" button is grayed out and appears pressed, giving the impression that it is also pressed

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.27.6

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/15fa91f6-3e26-48cf-8380-b6e2df5101c2

https://github.com/Expensify/App/assets/93399543/d369f876-3a7e-44e9-9e99-52c856c7e1e9

**Expensify/Expensify Issue URL:**

**Issue reported by:** @ayazhussain79

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686222783852909

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d97c2de3b309cd18</li>
        <li>Upwork Job ID: 1669708795862142976</li>
        <li>Last Price Increase: 2023-06-16</li>
    </ul>
</details>

**Comment by ahmedGaber93** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
"Mark as done" button is grayed out and appears pressed when clicking on "To" to assign a task
### What is the root cause of that problem?
Button Mark As Done is inside is in the clickable `PressableWithFeedback` that open Assignee modal, and the `PressableWithFeedback` has `pressDimmingValue={0.2}` which make the whole `PressableWithFeedback` has opacity when clicked.
https://github.com/Expensify/App/blob/4343d287ce1fff5bd1c4112c379fe645ba8650fc/src/components/TaskHeader.js#L59
### What changes do you think we should make in order to solve the problem?
Change the view structure to make button Mark As Done outside the clickable `PressableWithFeedback`.
the new structure will be

```javascript
<View style={[styles.flexRow]}>
    <PressableWithFeedback
        onPress={() => Navigation.navigate(ROUTES.getTaskReportAssigneeRoute(props.report.reportID))}
        wrapperStyle={[styles.flex1]}
        ....
    >
        <AssigneeAvatarAndNameIfFound/>
        // <we can also add placeholder text, to tell user to click here to assign>
   </PressableWithFeedback>
   
   <MarkAsREadButton>
</View>  
```

![Screenshot 2023-06-15 at 12 58 46 AM](https://github.com/Expensify/App/assets/41129870/6ecc4a59-b729-4f5c-b972-ebfe7c2b033e)

when press in

![Screenshot 2023-06-15 at 1 21 56 AM](https://github.com/Expensify/App/assets/41129870/6bb200d0-3712-4013-9807-bca4146862ec)



<details>
<summary>full code for test</summary>

    return (
        <View style={styles.borderBottom}>
            <View style={[{backgroundColor: themeColors.highlightBG}, styles.pl0]}>
                <View style={[styles.ph5, styles.pb5]}>
                    <Text style={[styles.textLabelSupporting, styles.lh16]}>{props.translate('common.to')}</Text>
                    <View style={[styles.flexRow]}>
                        <PressableWithFeedback
                            onPress={() => Navigation.navigate(ROUTES.getTaskReportAssigneeRoute(props.report.reportID))}
                            disabled={!isOpen}
                            accessibilityRole="button"
                            accessibilityLabel={props.translate('newTaskPage.assignee')}
                            hoverDimmingValue={1}
                            pressDimmingValue={0.2}
                            wrapperStyle={[styles.flex1]}
                        >
                            <View style={[styles.flexRow, styles.alignItemsCenter, styles.pv3]}>
                                {!_.isEmpty(props.report.managerEmail) && (
                                    <>
                                        <Avatar
                                            source={assigneeAvatar}
                                            type={CONST.ICON_TYPE_AVATAR}
                                            name={assigneeName}
                                            size={CONST.AVATAR_SIZE.HEADER}
                                        />
                                        <View style={[styles.flexColumn, styles.ml3]}>
                                            <Text
                                                style={[styles.headerText, styles.pre]}
                                                numberOfLines={1}
                                            >
                                                {assigneeName}
                                            </Text>
                                        </View>
                                    </>
                                )}
                            </View>

                        </PressableWithFeedback>

                        <View style={[styles.flexRow, styles.pv3]}>
                            {isCompleted ? (
                                <>
                                    <Text>{props.translate('task.completed')}</Text>
                                    <View style={styles.moneyRequestHeaderCheckmark}>
                                        <Icon
                                            src={Expensicons.Checkmark}
                                            fill={themeColors.iconSuccessFill}
                                        />
                                    </View>
                                </>
                            ) : (
                                <Button
                                    success
                                    isDisabled={TaskUtils.isTaskCanceled(props.report)}
                                    medium
                                    text={props.translate('newTaskPage.markAsDone')}
                                    onPress={() => TaskUtils.completeTask(props.report.reportID, title)}
                                />
                            )}
                        </View>

                    </View>
                </View>
            </View>
            <MenuItemWithTopDescription
                shouldShowHeaderTitle
                title={props.report.reportName}
                description={props.translate('newTaskPage.task')}
                onPress={() => Navigation.navigate(ROUTES.getTaskReportTitleRoute(props.report.reportID))}
                disabled={!isOpen}
                interactive={isOpen}
            />
            <MenuItemWithTopDescription
                title={lodashGet(props.report, 'description', '')}
                description={props.translate('newTaskPage.description')}
                onPress={() => Navigation.navigate(ROUTES.getTaskReportDescriptionRoute(props.report.reportID))}
                disabled={!isOpen}
                interactive={isOpen}
            />
        </View>
    );

</details>


### What alternative solutions did you explore? (Optional)
set `pressDimmingValue={1}`

**Comment by thienlnam** [comment]
We're redesigning the task view in https://github.com/Expensify/App/issues/19330, so let's not do anything there

**Comment by thienlnam** [comment]
Yeah I mean this issue - I'll just put it on hold so we don't get any more bug reports for this and then we can close it once the other PR is done!

**Comment by chiragsalian** [comment]
Issue is on HOLD, i'll demote it to weekly so that melvin does not ping us about it being overdue often.

**Comment by sonialiap** [comment]
https://github.com/Expensify/App/issues/19330 has been completed. Taking our issue off Hold

@thienlnam shall we close this issue?

**Comment by thienlnam** [comment]
Yup! I think if we can't reproduce this anymore we can close!

**Comment by rushatgabhane** [comment]
Awesome, let's close this.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)
